dask/typing.py

```diff
@@ -0,0 +1,364 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Callable, Hashable, Mapping, Sequence
+from typing import Any, Protocol, TypeVar, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
```

Comment:
`IPython` is sometimes an expensive import, which is probably why it's currently not imported until it's needed. This would make us import `IPython` at dask import time whenever it's installed.

It might be worth wrapping this inside of an `if TYPE_CHECKING` block to avoid incurring that runtime cost.

Reply:
Ah yes good idea, done.